src/onegov/api/models.py

```diff
@@ -107,67 +117,78 @@ class ApiEndpoint:
 
     """
 
-    name = ''
+    name: str = ''
+    endpoint: str = ''
     filters: list[str] = []
 
-    def __init__(self, app, extra_parameters=None, page=None):
+    def __init__(
+        self,
+        app: 'Framework',
+        extra_parameters: dict[str, 'str | int | None'] | None = None,
+        page: int | None = None,
+    ):
         self.app = app
         self.extra_parameters = extra_parameters or {}
         self.page = int(page) if page else page
         self.batch_size = 100
 
-    def for_page(self, page):
+    def for_page(self, page: int | None) -> 'Self | None':
         """ Return a new endpoint instance with the given page while keeping
         the current filters.
 
         """
 
         return self.__class__(self.app, self.extra_parameters, page)
 
-    def for_filter(self, **filters):
+    def for_filter(self, **filters: Any) -> 'Self':
         """ Return a new endpoint instance with the given filters while
         discarding the current filters and page.
 
         """
 
         return self.__class__(self.app, filters)
 
-    def for_item(self, item):
+    def for_item(self, item: _M | None) -> 'ApiEndpointItem[_M] | None':
```

Comment:
```suggestion
    @overload
    def for_item(self, None) ->  None: ...
    @overload
    def for_item(self, item: _M) -> 'ApiEndpointItem[_M]': ...
   
    def for_item(self, item: _M | None) -> 'ApiEndpointItem[_M] | None':
```
This is how you let mypy infer the return type a bit better from the passed in value